### ISSUE TITLE
Error when trying to pull public CRC Synapse table

### ISSUE THREAD

**Issue by janickweberpals** [GENIE-BPC/genieBPC#116]
Thank you for building this fantastic package!

I tried to pull the public CRC synapse tables via
`crc <- pull_data_synapse(cohort = "CRC", version = "v2.0-public")`

which led to the below error

```
Error in `purrr::map()`:
ℹ In index: 1.
ℹ With name: CRC_v2.0.
Caused by error in `dplyr::mutate()`:
ℹ In argument: `file_info = map(...)`.
Caused by error in `map()`:
ℹ In index: 1.
Caused by error in `switch()`:
! EXPR must be a length 1 vector
```
Pulling NSCLC works fine, so I think there must be an error in the function trying to pull the CRC tables.

Many thanks!

**Comment by hfuchs5** [comment]
![image](https://github.com/GENIE-BPC/genieBPC/assets/100386516/37f1df04-f39b-46fd-8bdc-b5678744f124)
Also an issue for BrCa v1.2-consortium

**Comment by jalavery** [comment]
> Thank you for building this fantastic package!
> 
> I tried to pull the public CRC synapse tables via `crc <- pull_data_synapse(cohort = "CRC", version = "v2.0-public")`
> 
> which led to the below error
> 
> ```
> Error in `purrr::map()`:
> ℹ In index: 1.
> ℹ With name: CRC_v2.0.
> Caused by error in `dplyr::mutate()`:
> ℹ In argument: `file_info = map(...)`.
> Caused by error in `map()`:
> ℹ In index: 1.
> Caused by error in `switch()`:
> ! EXPR must be a length 1 vector
> ```
> 
> Pulling NSCLC works fine, so I think there must be an error in the function trying to pull the CRC tables.
> 
> Many thanks!

Thanks for reporting this issue! Could you please include the version of the package that you are using? We are looking into this and will report back. Thank you!

**Comment by janickweberpals** [comment]
Thank you very much, it's genieBPC version 1.1.0

Here is the sessionInfo:

```
R version 4.2.1 (2022-06-23)

Platform: x86_64-apple-darwin17.0 (64-bit)

locale: en_US.UTF-8||en_US.UTF-8||en_US.UTF-8||C||en_US.UTF-8||en_US.UTF-8

attached base packages: stats, graphics, grDevices, datasets, utils, methods and base

other attached packages: gnomeR(v.1.3.0), genieBPC(v.1.1.0), smdi(v.0.2.2), ggsurvfit(v.0.3.0), survival(v.3.5-7), DT(v.0.28), gtsummary(v.1.7.2), readxl(v.1.4.3), gt(v.0.9.0), here(v.1.0.1), lubridate(v.1.9.2), forcats(v.1.0.0), stringr(v.1.5.0), dplyr(v.1.1.2), purrr(v.1.0.2), readr(v.2.1.4), tidyr(v.1.3.0), tibble(v.3.2.1), ggplot2(v.3.4.3) and tidyverse(v.2.0.0)

loaded via a namespace (and not attached): backports(v.1.4.1), circlize(v.0.4.15), plyr(v.1.8.8), splines(v.4.2.1), crosstalk(v.1.2.0), digest(v.0.6.31), foreach(v.1.5.2), htmltools(v.0.5.4), fansi(v.1.0.4), magrittr(v.2.0.3), cluster(v.2.1.3), doParallel(v.1.0.17), tzdb(v.0.4.0), ComplexHeatmap(v.2.14.0), dtplyr(v.1.3.1), matrixStats(v.1.0.0), timechange(v.0.2.0), colorspace(v.2.1-0), mitools(v.2.4), haven(v.2.5.3), pan(v.1.8), xfun(v.0.39), crayon(v.1.5.2), jsonlite(v.1.8.4), lme4(v.1.1-34), iterators(v.1.0.14), glue(v.1.6.2), gtable(v.0.3.3), GetoptLong(v.1.0.5), shape(v.1.4.6), BiocGenerics(v.0.44.0), jomo(v.2.7-6), scales(v.1.2.1), DBI(v.1.1.3), GGally(v.2.1.2), Rcpp(v.1.0.11), visR(v.0.3.1), clue(v.0.3-64), stats4(v.4.2.1), survey(v.4.2-1), glmnet(v.4.1-7), htmlwidgets(v.1.6.2), httr(v.1.4.6), RColorBrewer(v.1.1-3), ellipsis(v.0.3.2), mice(v.3.16.0), pkgconfig(v.2.0.3), reshape(v.0.8.9), farver(v.2.1.1), nnet(v.7.3-17), sass(v.0.4.5), janitor(v.2.2.0), utf8(v.1.2.3), tidyselect(v.1.2.0), labeling(v.0.4.2), rlang(v.1.1.1), munsell(v.0.5.0), cellranger(v.1.1.0), tools(v.4.2.1), cachem(v.1.0.7), cli(v.3.6.1), generics(v.0.1.3), broom(v.1.0.5), evaluate(v.0.20), fastmap(v.1.1.1), yaml(v.2.3.7), knitr(v.1.42), pander(v.0.6.5), randomForest(v.4.7-1.1), mitml(v.0.4-5), visdat(v.0.6.0), nlme(v.3.1-157), tictoc(v.1.2), xml2(v.1.3.5), compiler(v.4.2.1), rstudioapi(v.0.15.0), curl(v.5.0.2), png(v.0.1-8), broom.helpers(v.1.14.0), bslib(v.0.4.2), stringi(v.1.7.12), lattice(v.0.20-45), Matrix(v.1.4-1), commonmark(v.1.9.0), markdown(v.1.7), nloptr(v.2.0.3), vctrs(v.0.6.3), pillar(v.1.9.0), lifecycle(v.1.0.3), BiocManager(v.1.30.22), jquerylib(v.0.1.4), GlobalOptions(v.0.1.2), cowplot(v.1.1.1), data.table(v.1.14.8), corpcor(v.1.6.10), patchwork(v.1.1.3), R6(v.2.5.1), renv(v.1.0.2), IRanges(v.2.32.0), sessioninfo(v.1.2.2), codetools(v.0.2-18), d3r(v.1.0.1), boot(v.1.3-28), fastDummies(v.1.7.3), MASS(v.7.3-57), Hotelling(v.1.0-8), rprojroot(v.2.0.3), rjson(v.0.2.21), withr(v.2.5.0), S4Vectors(v.0.36.2), naniar(v.1.0.0), parallel(v.4.2.1), hms(v.1.1.3), sunburstR(v.2.1.8), labelled(v.2.12.0), grid(v.4.2.1), rpart(v.4.1.16), tableone(v.0.13.2), minqa(v.1.2.5), snakecase(v.0.11.0), rmarkdown(v.2.24) and pROC(v.1.18.4)
```

**Comment by jalavery** [comment]
@janickweberpals Thank you! Would you please install the latest dev version of the package via `remotes::install_github("GENIE-BPC/genieBPC`? This should resolve the issue, but please let us know if it doesn't. 

A note for our R package team: In the genieBPC v1.1.0 release there was a bug where the synapse table IDs for CRC 2.0-public were pointing to the copy of the CRC public release in the GENIE BPC Consortium section of Synapse as opposed to the GENIE BPC Public section of Synapse (this was the issue that Xiwen had when she was trying to access the CRC data). This was resolved in PR #107. I think that the potential issue with the synapse table IDs that Hannah is printing out above is separate.

**Comment by janickweberpals** [comment]
@jalavery Thank you very much for the fast patch! I confirm the dev version of the package works and I was able to pull the CRC data.

Thanks again so much!
